### PR TITLE
Update verifiers to rely on subset of evidence

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously the `AttributesVerifier` and `ConfigIdVerifier` were being
provided the parent structure and looking up the values themselves from
this parent structure. Now these types are provided the specific piece
of evidence they care about.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/2" target="_blank">Graphite</a> 
      * **PR #3** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/3" target="_blank">Graphite</a> 
        * **PR #9** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/9" target="_blank">Graphite</a> 
          * **PR #10** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/10" target="_blank">Graphite</a> 
            * **PR #11** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/11" target="_blank">Graphite</a> 
              * **PR #12** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/12" target="_blank">Graphite</a>  👈
    * **PR #4** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/4" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/12?utm_source=stack-comment).

File: verifier/src/report_body.rs [hunk 1]
```diff
@@ -117,11 +117,11 @@ impl ConfigIdVerifier {
...
     }
 }
-impl Verifier<ReportBody> for ConfigIdVerifier {
+impl Verifier<ConfigId> for ConfigIdVerifier {
     type Error = VerificationError;
-    fn verify(&self, evidence: &ReportBody) -> CtOption<Self::Error> {
+    fn verify(&self, evidence: &ConfigId) -> CtOption<Self::Error> {
         let expected = self.expected_id.clone();
-        let actual = evidence.config_id();
+        let actual = evidence.clone();
```
Comment (nick-mobilecoin) [inline verifier/src/report_body.rs:124]: Not fond of this cloning multiple times, but I think the use case of the verifier is narrow enough that the extra cloning of the data isn't likely to be an issue for the entire system

File: verifier/src/report_body.rs [hunk 2]
```diff
@@ -221,37 +222,39 @@ mod test {
 
     #[test]
     fn attributes_success() {
-        let report_body = ReportBody::from(&REPORT_BODY_SRC);
-        let verifier = AttributesVerifier::new(REPORT_BODY_SRC.attributes.into());
+        let attributes = Attributes::default()
+            .set_flags(1)
```
Comment (samdealy) [inline verifier/src/report_body.rs:226]: Does `1` refer to a specific flag? If so, I'd recommend using a variable that names the flag being used.

File: verifier/src/report_body.rs [hunk 3]
```diff
@@ -221,37 +222,39 @@ mod test {
...
+            .set_extended_features_mask(2);
```
Comment (samdealy) [inline verifier/src/report_body.rs:227]: Same as above.

**Review by samdealy — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
With the refactor of #11 this is no longer needed

**Pull request closed without merging**